File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/geometry/CoordinateFormatterController.java
package com.valentingrigorean.arcgis_maps_flutter.geometry;

import androidx.annotation.NonNull;


import com.esri.arcgisruntime.geometry.CoordinateFormatter;
import com.esri.arcgisruntime.geometry.Point;
import com.valentingrigorean.arcgis_maps_flutter.Convert;

import java.util.Map;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;

public class CoordinateFormatterController implements MethodChannel.MethodCallHandler {
    private static final String TAG = "CoordinateFormatterController";
    private final MethodChannel channel;

    public CoordinateFormatterController(BinaryMessenger messenger) {
        channel = new MethodChannel(messenger, "plugins.flutter.io/arcgis_channel/coordinate_formatter");
        channel.setMethodCallHandler(this);
    }

    public void dispose() {
        channel.setMethodCallHandler(null);
    }

    @Override
    public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        switch (call.method) {
            case "latitudeLongitudeString":
                final Map<?, ?> data = Convert.toMap(call.arguments);
                final Point from = (Point) Convert.toGeometry(data.get("from"));
                final int format = Convert.toInt(data.get("format"));
                final int decimalPlaces = Convert.toInt(data.get("decimalPlaces"));
                result.success(CoordinateFormatter.toLatitudeLongitude(from, CoordinateFormatter.LatitudeLongitudeFormat.values()[format], decimalPlaces));
                break;
            default:
                result.notImplemented();
                break;
        }
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/geometry/CoordinateFormatterController.kt
package com.valentingrigorean.arcgis_maps_flutter.geometry

import com.arcgismaps.geometry.CoordinateFormatter
import com.arcgismaps.geometry.GeodesicSectorParameters
import com.arcgismaps.geometry.Point
import com.valentingrigorean.arcgis_maps_flutter.convert.geometry.toGeometryOrNull
import com.valentingrigorean.arcgis_maps_flutter.convert.geometry.toLatitudeLongitudeFormat
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class CoordinateFormatterController(messenger: BinaryMessenger) : MethodCallHandler {
    private val channel: MethodChannel

    init {
        channel =
            MethodChannel(messenger, "plugins.flutter.io/arcgis_channel/coordinate_formatter")
        channel.setMethodCallHandler(this)
    }

    fun dispose() {
        channel.setMethodCallHandler(null)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "latitudeLongitudeString" -> {
                val data = call.arguments as Map<*, *>
                val from =  data["from"]?.toGeometryOrNull()!! as Point
                val format = (data["format"] as Int).toLatitudeLongitudeFormat()
                val decimalPlaces = data["decimalPlaces"] as Int
                result.success(
                    CoordinateFormatter.toLatitudeLongitudeOrNull(
                        from,
                        format,
                        decimalPlaces
                    )
                )
            }

            else -> result.notImplemented()
        }
    }
}